casper/src/test/scala/coop/rchain/casper/MultiParentCasperMergeSpec.scala

```diff
@@ -53,33 +59,152 @@ class MultiParentCasperMergeSpec extends FlatSpec with Matchers with Inspectors
   }
 
   it should "handle multi-parent blocks correctly when they operate on stdout" ignore effectTest {
-    def echoContract(no: Int) = s"""new stdout(`rho:io:stdout`) in { stdout!("Contract $no") }"""
-    val time                  = System.currentTimeMillis()
-    HashSetCasperTestNode.networkEff(genesis, networkSize = 2).use { nodes =>
-      val deploys = Vector(
-        ConstructDeploy.sourceDeploy(echoContract(1), timestamp = time + 1),
-        ConstructDeploy.sourceDeploy(echoContract(2), timestamp = time + 2)
-      )
-      for {
-        block0 <- nodes(0).addBlock(deploys(0))
-        block1 <- nodes(1).addBlock(deploys(1))
-        _      <- nodes(0).receive()
-        _      <- nodes(1).receive()
-        _      <- nodes(0).receive()
-        _      <- nodes(1).receive()
+    def echoContract(no: Int) =
+      Rho(s"""new stdout(`rho:io:stdout`) in { stdout!("Contract $no") }""")
+    merges(echoContract(1), echoContract(2), Rho("Nil"))
+  }
 
-        //multiparent block joining block0 and block1 since they do not conflict
-        multiparentBlock <- nodes(0).addBlock(deploys(1))
-        _                <- nodes(1).receive()
+  // TODO: Peek rows/column
+  // Note this skips pairs that lead to infinite loops
+  it should "handle multi-parent blocks correctly when they operate on volatile produce/consume pairs" in effectTest {
+    // Sends (linear sends)
+    val S0 = Rho("@0!(0)")
+    val S1 = Rho("@0!(1)")
+    // Repeats (persistent sends)
+    val R0 = Rho("@0!!(0)")
+    val R1 = Rho("@0!!(1)")
+    // For-s (linear receives)
+    val F_ = Rho("for (_ <- @0) { 0 }")
+    val F0 = Rho("for (@0 <- @0) { 0 }")
+    val F1 = Rho("for (@1 <- @0) { 0 }")
+    // Contracts (persistent receives)
+    val C_ = Rho("contract @0(id) = { 0 }")
+    val C0 = Rho("contract @0(@0) = { 0 }")
+    val C1 = Rho("contract @0(@1) = { 0 }")
+    //FIXME add missing cases for in-deploy COMM-s wherever there's a pair without X (a COMM)
+    //FIXME all `conflictsForNow` should eventually be replaced with `merges`
+    Map(
+      "!X !X"        -> conflictsForNow(S0, S0, Nil),
```

Comment:
it would be so cool to have an additional field with reference to the reason why it conflicts for now; ticket, page, discord discussion, anything that would allow someone not familiar with the (large) problem space to catch up and see what needs to be done to enable this.

Reply:
The JIRA ticket would be a better place than code

Reply:
exactly my point. link to a spot that describes what needs to change so the bus factor is >1.